clms-api/src/main/java/com/clms/api/assignments/Assignment.java

```diff
@@ -5,13 +5,16 @@
 import com.fasterxml.jackson.annotation.JsonIgnore;
 import jakarta.persistence.*;
 import lombok.Data;
+import lombok.Getter;
+import lombok.Setter;
 
 import java.util.Date;
 import java.util.List;
 
 @Entity
 @Table(name = "assignments")
-@Data
+@Getter
+@Setter
 public class Assignment {
 
     @Id
```

Comment:
@Data annotation was automatically generating toString methods which was causing circular dependency of loading assignment -> course -> assignment -> course -> ..., I think this is a good reason to use projections to transfer data back to the clients (https://docs.spring.io/spring-data/jpa/reference/repositories/projections.html)